064-topic-replication-factor-change.md

```diff
@@ -0,0 +1,291 @@
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=2&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "5344ca89-351f-4565-8d0f-9aade00e053d",
+      "StartMs": "1701961135978"
+    }
+  ],
+  "version": 1
+}
+```
+
+### Configuration parameters
+
+The Topic Operator needs to authenticate and send HTTP requests to the Cruise Control's REST API.
+The following environment variables are set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
+
+| Name                                  | Type    | Default | Description                                                   |
+|:--------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED        | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED   | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME       | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT           | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED    | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED   | Boolean | false   | Whether Basic authentication is enabled                       |
+| STRIMZI_CRUISE_CONTROL_AUTH_USERNAME  | String  | ""      | Username of the REST API user with admin role                 |
+| STRIMZI_CRUISE_CONTROL_AUTH_PASSWORD  | String  | ""      | Password of the REST API user with admin role                 |
+
+With Cluster Operator deployment, the HTTP client truststore is generated in the Topic Operator pod, and a dedicated Cruise Control admin user is created.
+With standalone deployment, the user has to mount the certificate chain in PEM format under `/etc/tls-sidecar/cluster-ca-certs/ca.crt`, and specify the admin user credentials using the environment variables.
+
+### KafkaTopic CRD changes
+
+During the whole replicas change process the KafkaTopic remains in a ready state.
+A replicas change is executed asynchronously, and takes more than one reconciliation to complete.
+
+A new optional KafkaTopicStatus subsection called `replicasChange` is used to update the user, and keep track of the replicas change state.
+The `replicasChange` can be in a pending or ongoing `state`, and it can contains a `message` in case of failure.
+Other fields are `sessionId`, which maps to Cruise Control's `User-Task-ID`, and `replicas`, which reflects the target replicas value (this may be different from .spec.replicas when the state is ongoing).
```

Comment:
what about `replicasChange.target` or `replicasChange.targetReplicas` instead of just `replicas`?
Or what about having something like ...

```yaml
status:
  replicas:
    current: <current resplicas from spec>
    target: <n. replicas requested to change to>
    state: pending|ongoing
```

Reply:
I like `targetReplicas`.